test/functional-tests-legacy/CMakeLists.txt

```diff
@@ -26,41 +26,49 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-if (BUILD_TESTING)
+if(BUILD_TESTING)
+    if("${CMAKE_CURRENT_BINARY_DIR}" MATCHES ".* .*")
+        message(WARNING
+            "The build directory path contains a space. Unfortunately, the"
+            " legacy functional tests are known not to work in such a case."
+            " For this reason, they will be deactivated for this build.")
+        return()
```

Comment:
I strongly disapprove on the use of `return()` here. This is not returning from the current block but rather from the hole file. Meaning that an subsequent include would not be evaluated. I do not think anybody expect non linear evaluation in cmakfiles.